chanzuckerberg/czgenepi
# thuang-hsts-csp-135163

### Description

1. Add `build` folder in `src/frontend/.stylelintignore`
2. Replace the usage of npm package - `serve` as our frontend server with Express, so we can add the security headers
3. Add HSTS header with 2 years TTL
4. Add CSP that allow all the resources we currently use to load properly. However, CSP is a living policy, which means whenever we add new resources (Google Analytics, etc.) we'll need to update our allowlist accordingly. Likely [Meta's CSP setting](https://github.com/chanzuckerberg/khepri/blob/develop/src/server/common/cspSettings.ts) will have examples we need down the road

BONUS:
1. Update tree modal outlink to Nextstrain.org as opening in a new tab instead of the current tab
2. Update agree ToS "Decline" button to redirect instead of making an API call to logout, which didn't work 

#### Issue
https://app.clubhouse.io/genepi/story/135163/frontend-security-add-hsts-and-csp

### Test plan

1. Visit https://thuang-hsts-1-frontend.dev.genepi.czi.technology/data/samples
2. Check the html request's response header and you should see content security policy and strict transport security headers in place!
3. 
<img width="1564" alt="Screen Shot 2021-05-05 at 11 59 58 AM" src="https://user-images.githubusercontent.com/6309723/117195019-a08b0900-ad99-11eb-87bb-75fc9cdc4366.png">

PTAL thank you!
